aws/resource_aws_autoscaling_policy.go

```diff
@@ -71,6 +61,137 @@ func resourceAwsAutoscalingPolicy() *schema.Resource {
+													Required: true,
+													ValidateFunc: validation.StringInSlice([]string{
+														"ASGCPUUtilization",
+														"ASGNetworkIn",
+														"ASGNetworkOut",
+														"ALBRequestCount",
+													}, false),
+												},
+												"resource_label": {
+													Type:     schema.TypeString,
+													Required: true,
+												},
+											},
+										},
+									},
+									"predefined_scaling_metric_specification": {
+										Type:     schema.TypeList,
+										Optional: true,
+										MaxItems: 1,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"predefined_metric_type": {
+													Type:     schema.TypeString,
+													Required: true,
+													ValidateFunc: validation.StringInSlice([]string{
+														"ASGAverageCPUUtilization",
+														"ASGAverageNetworkIn",
+														"ASGAverageNetworkOut",
+														"ALBRequestCountPerTarget",
+													}, false),
+												},
+												"resource_label": {
+													Type:     schema.TypeString,
+													Required: true,
+												},
+											},
+										},
+									},
+									"predefined_load_metric_specification": {
+										Type:     schema.TypeList,
+										Optional: true,
+										MaxItems: 1,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"predefined_metric_type": {
+													Type:     schema.TypeString,
+													Required: true,
+													ValidateFunc: validation.StringInSlice([]string{
+														"ASGTotalCPUUtilization",
+														"ASGTotalNetworkIn",
+														"ASGTotalNetworkOut",
+														"ALBTargetGroupRequestCount",
+													}, false),
+												},
+												"resource_label": {
+													Type:     schema.TypeString,
+													Required: true,
+												},
+											},
+										},
+									},
+									"target_value": {
+										Type:     schema.TypeInt,
+										Required: true,
+									},
+								},
+							},
+						},
+						"max_capacity_breach_behavior": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Default:  "HonorMaxCapacity",
+							ValidateFunc: validation.StringInSlice([]string{
+								"HonorMaxCapacity",
+								"IncreaseMaxCapacity",
+							}, false),
+						},
+						"max_capacity_buffer": {
+							Type:         schema.TypeInt,
+							Optional:     true,
+							Default:      0,
+							ValidateFunc: validation.IntBetween(-1, 100),
+						},
+						"mode": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Default:  "ForecastOnly",
+							ValidateFunc: validation.StringInSlice([]string{
+								"ForecastOnly",
+								"ForecastAndScale",
+							}, false),
+						},
+						"scheduling_buffer_time": {
+							Type:         schema.TypeInt,
+							Optional:     true,
+							ValidateFunc: validation.IntAtLeast(0),
```

Comment:
This should also be a `nullable.Int` https://github.com/hashicorp/terraform-provider-aws/blob/main/aws/internal/experimental/nullable/int.go, since Terraform defaults to treating the zero-value as unset, whereas the AWS API defaults this value to 300 if not set.

There should be an acceptance test to validate that the API returns a value of `0` if `scheduling_buffer_time = 0` and returns `300` if it is not set.